src/screens/ParadeMapScreen/Map.js

```diff
@@ -0,0 +1,192 @@
+// @flow
+import React, { Component } from "react";
+import { StyleSheet, View } from "react-native";
+import MapView, { Polyline, Marker } from "react-native-maps";
+import type { Event, SavedEvents } from "../../data/event";
+import paradeCoordinates from "../../constants/parade-coordinates";
+import {
+  warmPinkColor,
+  lightNavyBlueColor,
+  transparent
+} from "../../constants/colors";
+import Text, { scaleWithFont } from "../../components/Text";
+import EventCard from "../../components/EventCard";
+import ContentPadding from "../../components/ContentPadding";
+import stageIconActive from "../../../assets/images/stageIconActive.png";
+import stageIconInactive from "../../../assets/images/stageIconInactive.png";
+
+type Props = {
+  stages: Array<Event>,
+  savedEvents: SavedEvents,
+  addSavedEvent: string => void,
+  removeSavedEvent: string => void,
+  onPress: (id: string) => void
+};
+
+type State = {
+  activeMarker: ?string,
+  tileDetails: ?Event
+};
+
+class Map extends Component<Props, State> {
+  state = {
+    activeMarker: null,
+    tileDetails: null
+  };
+
+  handleMarkerPress(stage: Event) {
+    this.setState({ tileDetails: stage, activeMarker: stage.id });
+  }
+
+  handleMapPress = () => {
+    this.setState({ tileDetails: null, activeMarker: null });
+  };
+
+  renderStageMarker = (stage: Event) => (
+    <Marker
+      coordinate={{
+        longitude: stage.fields.location.lon,
+        latitude: stage.fields.location.lat
+      }}
+      key={stage.id}
+      onPress={() => this.handleMarkerPress(stage)}
+      stopPropagation
+      image={
+        this.state.activeMarker === stage.id
+          ? stageIconActive
+          : stageIconInactive
+      }
+    />
+  );
+
+  render() {
+    const {
+      savedEvents,
+      addSavedEvent,
+      removeSavedEvent,
+      onPress
+    } = this.props;
+    return (
+      <View style={styles.mapWrapper}>
+        <MapView
+          style={StyleSheet.absoluteFill}
+          initialRegion={{
+            latitude: 51.51004,
+            longitude: -0.134192,
+            latitudeDelta: 0.02,
+            longitudeDelta: 0.000002
+          }}
+          showsPointsOfInterest={false}
+          showsScale={false}
+          showsBuildings={false}
+          showsTraffic={false}
+          showsIndoors={false}
+          onPress={this.handleMapPress}
+        >
+          <Polyline
+            coordinates={paradeCoordinates}
+            strokeWidth={5}
+            strokeColor={warmPinkColor}
+            lineJoin="bevel"
+          />
+          <Marker
+            coordinate={{ longitude: -0.14223, latitude: 51.51616 }}
+            centerOffset={{ x: 0, y: -15 }}
+            stopPropagation
+          >
+            <View style={styles.markerView}>
```

Comment:
Should this be its own component? look to be repeated a few times here

Reply:
This repetition is reduced in the user location PR